presto-hive/src/main/java/com/facebook/presto/hive/util/HiveFileIterator.java

```diff
@@ -31,6 +31,7 @@
 import java.util.Collections;
 import java.util.Deque;
 import java.util.Iterator;
+import java.util.Optional;
```

Comment:
On a second thought it will be easier, if you can undo this change to this file.

Where you need the filter to always work, pass in path filter where it always returns true. () -> true

This will make it easier for me to merge this PR.

Reply:
PathFilter was added to DirectoryLister [due to Hudi](https://github.com/prestodb/presto/pull/13818/files#diff-d8caebd0319edf28ebbe49a5370a8a59f5bc767726f321ceab12e88ead8a2fc3). After this patch, we don't really need it. I was planning to remove that as a follow-up to this PR if that's ok with you?

Reply:
Let us do it with this PR itself.

Reply:
got it.. will update this PR in a day or two.

Reply:
updated the PR